src/controllers/components/cart-panel.js

```diff
@@ -0,0 +1,37 @@
+import { searchItem, checkIn} from '../../lib/api-client';
+import { Dispatcher } from 'consus-core/flux';
+import StudentStore from '../../store/student-store';
+
+export default class CartController {
+
+    static checkInItem(id, itemAddress) {
+        return checkIn(id, itemAddress).then(item => {
+            Dispatcher.handleAction('CHECKIN_SUCCESS', {
+                itemAddress: item.itemAddress
+            });
+        }).catch(error => {
+            Dispatcher.handleAction('ERROR', {
+                error
+            });
+        });
+    }
+
+    static getItem(address) {
+        if(StudentStore.getStudent().hasOverdueItem)
```

Comment:
Nit: I don't like the lack of `{ }` when the next statement is multi-line.

(pretend this is commented on all similar lines)